redwood-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/lazylayout/uiview/UIViewLazyList.kt

```diff
@@ -170,16 +181,12 @@ internal open class UIViewLazyList() : LazyList<UIView>, ChangeListener {
         layout: UICollectionViewLayout,
         sizeForItemAtIndexPath: NSIndexPath,
       ): CValue<CGSize> {
-        // TODO: Calculate size for individual items. This commented code currently hangs the UI Thread
-//        val item = items.itemForGlobalIndex(sizeForItemAtIndexPath.item.toInt())
-//        collectionView.frame().useContents {
-//          item.sizeThatFits(size.readValue()).useContents {
-//            return CGSizeMake(width, height)
-//          }
-//        }
-
-        // TODO: Handle size for horizontal scrollDirection
-        return CGSizeMake(collectionView.frame().useContents { size.width }, 64.0)
+        val itemSize = items.itemForGlobalIndex(sizeForItemAtIndexPath.item.toInt()).size
+        if (isVertical) {
+          return CGSizeMake(collectionView.frame().useContents { size.width }, itemSize.useContents { height })
+        } else {
+          return CGSizeMake(itemSize.useContents { width }, collectionView.frame().useContents { size.width })
+        }
```

Comment:
The vertical case uses the frame width and the item height but the horizontal case uses the item width and the frame width. Should the latter be item width and frame _height_?

Also, if you want, you can hoist the `return` outside the conditional.
```suggestion
        return if (isVertical) {
          CGSizeMake(collectionView.frame().useContents { size.width }, itemSize.useContents { height })
        } else {
          CGSizeMake(itemSize.useContents { width }, collectionView.frame().useContents { size.width })
        }
```

Reply:
Whoops yeah, meant to use height

Reply:
Fixed!